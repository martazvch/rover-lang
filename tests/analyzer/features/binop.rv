fn main() {
    _ = 1 + 2.
    _ = 5.6 + 4
    _ = "steak" * 3
    _ = 5 * "house"
    _ = "juicy " + "orange"
}
