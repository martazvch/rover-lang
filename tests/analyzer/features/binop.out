[Fn declaration main, index: 0, scope: Global, return kind: ImplicitVoid]
    [Discard]
        [Binop type: AddFloat, cast: Lhs]
            [Int 1]
            [Float 2]
    [Discard]
        [Binop type: AddFloat, cast: Rhs]
            [Float 5.6]
            [Int 4]
    [Discard]
        [Binop type: MulStr, cast: Rhs]
            [String steak]
            [Int 3]
    [Discard]
        [Binop type: MulStr, cast: Lhs]
            [Int 5]
            [String house]
    [Discard]
        [Binop type: AddStr, cast: None]
            [String juicy ]
            [String orange]
